### PR TITLE
describegpt: add --prompt-file option (resolves #1085)

### LINKED ISSUE #1085: `describegpt`: add a --prompt-file option
The prompt file is a json file specifying the dictionary prompt, description prompt and the tags prompt to use.

If specified, it will load the prompts from the json file instead of using the default prompts.

### PR DESCRIPTION
**Pull request by rzmk**
## 🗺 Overview

- Adds [`--prompt-file`](https://github.com/jqnatividad/qsv/blob/444ec4132ab959ef14f66e7a256da707d34d396a/src/cmd/describegpt.rs#L25) option to `describegpt` & a doc file at [`docs/Describegpt.md`](https://github.com/jqnatividad/qsv/blob/describegpt/prompt-file/docs/Describegpt.md).
- In `src/cmd/describegpt.rs`, refactors `get_dictionary_prompt`, `get_description_prompt`, & `get_tags_prompt` to functions `get_prompt_file` & `get_prompt`.

Resolves #1085.

### CODE REVIEW

File: src/cmd/describegpt.rs
Comment (rzmk) [inline src/cmd/describegpt.rs]: @jqnatividad Since `--max-tokens` has a default of 50, how can we check if the option itself is being used?

I'm trying to make it so that if `--max-tokens` is explicitly set then it overrides the prompt file `tokens` value like you mentioned [in the issue](https://github.com/jqnatividad/qsv/issues/1085#:~:text=the%20tokens%20attribute%20will%20be%20used%20unless%20%2D%2Dmax%2Dtokens%20option%20is%20set%20explicitly%20when%20invoking%20describegpt.).

Comment (jqnatividad) [reply src/cmd/describegpt.rs]: @rzmk , it's a bit hacky, but this is what I'd do:

```rust
pub fn run(argv: &[&str]) -> CliResult<()> {
  let args: Args = util::get_args(USAGE, argv)?;
  // simulate invoking describegpt with just a stdin input, and no options set, so we get the defaults
  let argv_defaults = &["describegpt", "-"];
  let args_defaults: Args = util::get_args(USAGE, argv_defaults)?;
```
so `args_defaults.flag_max_tokens` will contain the default value specified in the USAGE text, which you can use later in the code to see if it was changed.  This approach gives us the flexibility to insulate ourselves from future changes to the USAGE text as defaults, additional args/options are added.

Comment (rzmk) [reply src/cmd/describegpt.rs]: @jqnatividad Getting an error this way:

```console
$ ./target/debug/qsv describegpt NYC_311_SR_2010-2020-sample-1M.csv --tags --max-tokens 100
Invalid arguments.

Usage:
    qsv describegpt [options] [<input>]
    qsv describegpt --help
```

Also another two ways that may work are:

- Looping through `argv` for `--max-tokens` and getting the subsequent value.
- Making `--max-tokens` optional so we can use `is_some()`.

Both methods don't make `args_default` though which could be useful in the future.

Comment (jqnatividad) [reply src/cmd/describegpt.rs]: My mistake @rzmk , the first element of argv is the name of the binary:

```rust
pub fn run(argv: &[&str]) -> CliResult<()> {
  let args: Args = util::get_args(USAGE, argv)?;
  // simulate invoking describegpt with just a stdin input, and no options set, so we get the defaults
  let argv_defaults = &[argv[0], "describegpt", "-"];
  let args_defaults: Args = util::get_args(USAGE, argv_defaults)?;
```

Comment (rzmk) [reply src/cmd/describegpt.rs]: Still getting the same error @jqnatividad.

Comment (jqnatividad) [reply src/cmd/describegpt.rs]: Hhmmm....

It's working for me:
```
$ cargo t describegpt -F lite

...

$ target/debug/qsvlite describegpt C:\Users\joeln\Downloads\NYC_311_SR_2010-2020-sample-1M.csv --tags --max-tokens 100
Generating stats from \\?\C:\Users\joeln\Downloads\NYC_311_SR_2010-2020-sample-1M.csv using qsv stats --everything...
Generating frequency from \\?\C:\Users\joeln\Downloads\NYC_311_SR_2010-2020-sample-1M.csv using qsv frequency...
Interacting with OpenAI API...

Generating tags from OpenAI API...
Received tags completion.
uniquekey, createddate, closeddate, agency, agencyname, complainttype, descriptor, locationtype, incidentzip, incidentaddress, streetname, crossstreet1, crossstreet2, intersectionstreet1, intersectionstreet2, addressType, city, landmark, facilitytype, status, duedate, resolutiondescription, resolutionactionupdateddate, communityboard, bbl, borough, xcoordinatestateplane, ycoordinatestateplane, opendatachanneltype, parkfacilityname, park
```

Comment (rzmk) [reply src/cmd/describegpt.rs]: Somehow it works now for me?! 🎉

Comment (rzmk) [reply src/cmd/describegpt.rs]: @jqnatividad What if a user specifies `--max-tokens 50` then how can I differentiate between default? This is mainly what I'm considering because a user may have `7000` in their prompt file for `tokens` but then also set `--max-tokens 50`. How can I identify if `--max-tokens` is explicitly set so I can prioritize `--max-tokens` over the prompt file value if `args.flag_max_tokens == args_defaults.flag_max_tokens`?

Edit: I'm trying a closure right now.

File: src/cmd/describegpt.rs
Comment (rzmk) [inline src/cmd/describegpt.rs:478]: Here's a closure that can resolve the prioritization issue.

File: docs/Describegpt.md
Comment (jqnatividad) [inline docs/Describegpt.md:1]: @rzmk ❤️how thorough describegpt's documentation is.

**Review by jqnatividad — APPROVED** [review]
LGTM! Let's collab on it once its merged @rzmk

**Pull request merged**
